app/figures.py

```diff
@@ -110,3 +112,251 @@ def generate_system_freq_fig(df: pd.DataFrame) -> px.line:
             ],
         ).update_layout(yaxis_title="GW")
     return system_freq_fig
+
+
+def generate_intraday_market_sys_fig(df: pd.DataFrame) -> go:
+    """Creates Plotly figure for Intra-day Market System graph.
+
+    Args:
+        df: Opal data DataFrame
+
+    Returns:
+        Plotly graph_objects figure
+    """
+    intraday_market_sys_fig = make_subplots(specs=[[{"secondary_y": True}]])
+
+    if len(df.columns) == 1:
+        return intraday_market_sys_fig
+
+    # Left axis data
+    columns = [
```

Comment:
Really small style bugbear. Probably better to just name the variable `left_axis_columns` instead of using a comment and a vague variable name you overwrite later on
```suggestion
    left_axis_columns = [
```